cmd/agent/app/integrations.go

```diff
@@ -270,16 +319,35 @@ func installTuf(cmd *cobra.Command, args []string) error {
 	}
 
 	intVer := strings.Split(args[0], "==")
-	integration := strings.TrimSpace(intVer[0])
+	integration := strings.Replace(strings.TrimSpace(intVer[0]), "_", "-", -1)
 	if integration == "datadog-checks-base" {
-		return fmt.Errorf("cannot upgrade datadog-checks-base")
+		return fmt.Errorf("this command does not allow installing datadog-checks-base")
+	}
+	versionToInstall, err := parseVersion(strings.TrimSpace(intVer[1]))
+	if err != nil {
+		return fmt.Errorf("unable to get version of %s to install: %v", integration, err)
 	}
-	versionToInstall := strings.TrimSpace(intVer[1])
-	currentVersion, err := getIntegrationVersion(integration, cachePath)
+	currentVersion, err := installedVersion(integration, cachePath)
 	if err != nil {
 		return fmt.Errorf("could not get current version of %s: %v", integration, err)
 	}
 
+	if *versionToInstall == *currentVersion {
```

Comment:
you need to check `currentVersion` is not `nil` before dereferencing it (pretty sure the code segfaults right now if the integration is not already installed). Also, might be worth documenting on `installedVersion` that it can return a `nil` version if the integration's not installed :)

Also, nit, might be personal preference: for semantics, I'd prefer something like `versionToInstall.Equals(currentVersion)`. Up to you though, feel free to disregard.